RileyLink/PumpStatusMessage.m

```diff
@@ -25,39 +25,49 @@ - (instancetype)initWithData:(NSData*)data
 }
 
 - (NSDictionary*) bitBlocks {
-  return @{@"sequence": @[@1, @7],
-           @"trend": @[@12, @3],
-           @"sensor_minute": @[@234, @6],
-           @"sensor_hour": @[@227, @5],
-           @"sensor_day": @[@267, @5],
-           @"sensor_month": @[@260, @4],
-           @"sensor_year": @[@248, @8],
-           @"bg_h": @[@72, @8],
-           @"bg_l": @[@199, @1],
-           @"prev_bg_h": @[@80, @8],
-           @"prev_bg_l": @[@198, @1],
-//sequence: [1,7],
-//pump_hour: [19,5],
-//pump_minute: [26,6],
-//pump_year: [40,8],
-//pump_month: [52,4],
-//pump_day: [59,5],
-//bg_h: [72, 8],
-//prev_bg_h: [80, 8],
-//insulin_remaining: [101,11],
-//sensor_age: [144,8],
-//sensor_remaining: [152,8],
-           @"active_ins": @[@181, @11]
-//prev_bg_l: [198, 1],
-//bg_l: [199, 1],
-//sensor_hour: [227,5],
-//sensor_minute: [234,6],
-//sensor_year: [248, 8],
-//sensor_month: [260, 4],
-//sensor_day: [267,5]
+  return @{@"sequence":          @[@1, @7],
+           @"trend":             @[@12, @3],
+           @"bg_h":              @[@72, @8],
+           @"prev_bg_h":         @[@80, @8],
+           @"insulin_remaining": @[@101,@11],
+           @"batt":              @[@116,@4],
+           @"sensor_age":        @[@144,@8],
+           @"sensor_remaining":  @[@152,@8],
+           @"next_cal_hour":     @[@160,@8], // ff at sensor end, 00 at sensor off
+           @"next_cal_minute":   @[@168,@8],
+           @"active_ins":        @[@181, @11],
+           @"prev_bg_l":         @[@198, @1],
+           @"bg_l":              @[@199, @1],
+           @"sensor_hour":       @[@227, @5],
+           @"sensor_minute":     @[@234, @6],
+           @"sensor_year":       @[@248, @8],
+           @"sensor_month":      @[@260, @4],
+           @"sensor_day":        @[@267, @5],
   };
 }
 
+- (NSString*) nextCal {
```

Comment:
If you wanted to generate the absolute NSDate, you could use the awesome `NSCalendar` method `nextDateAfterDate` using the pump/sensor date. [Here it is in Swift as a reference](https://github.com/loudnate/naterade-ios/blob/master/MinimedKit/Messages/MySentryPumpStatusMessageBody.swift#L159-L164)